File: leetcode/1018. Binary Prefix Divisible By 5/1018.java
// https://helloacm.com/binary-prefix-divisible-by-5-java-c-coding-exercise/
// https://leetcode.com/problems/binary-prefix-divisible-by-5/

class Solution {
    public List<Boolean> prefixesDivBy5(int[] A) {
        List<Boolean> r = new ArrayList<>();
        int num = 0;
        for (int x: A) {
            num = ((num << 1) + x) % 5;
            r.add(num % 5 == 0);
        }        
        return r;
    }
}

File: leetcode/1018. Binary Prefix Divisible By 5/1018.cpp
// https://helloacm.com/binary-prefix-divisible-by-5-java-c-coding-exercise/
// https://leetcode.com/problems/binary-prefix-divisible-by-5/

class Solution {
public:
    vector<bool> prefixesDivBy5(vector<int>& A) {
        vector<bool> result;
        int x = 0;
        for (const auto y: A) {
            x = ((x << 1) + y) % 5;
            result.push_back(x % 5 == 0);
        }
        return result;
    }
};
